### PR TITLE
Code monitoring: Add manage page functionality

### PR DESCRIPTION
**Pull request by attfarhan**
Stacked on https://github.com/sourcegraph/sourcegraph/pull/16440. Adds functionality for the code monitor manage/edit page. Updates the CodeMonitorForm accept a code monitor via props to initialize the form. 

Still TODO in follow-up PRs:
- Update/factor out CSS
- Add different conditions for disabling the continue/cancel buttons on the manage page.

### CODE REVIEW

File: client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx [hunk 1]
```diff
@@ -1,3 +1,115 @@
...
+            <PageHeader title="Manage code monitor" />
+            Code monitors watch your code for specific triggers and run actions in response.{' '}
+            <a href="" target="_blank" rel="noopener">
+                {/* TODO: populate link */}
+                Learn more
+            </a>
+            {codeMonitorOrError === 'loading' && <LoadingSpinner className="icon-inline" />}
+            {codeMonitorOrError && !isErrorLike(codeMonitorOrError) && codeMonitorOrError !== 'loading' && (
+                <>
+                    <CodeMonitorForm
```
Comment (limitedmage) [inline client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:105]: When I open this page, clicking "Edit" on the trigger always shows the search query as invalid initially, even when it is valid.

File: client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx [hunk 2]
```diff
@@ -1,3 +1,115 @@
...
+import { CodeMonitoringProps } from '.'
+import { Scalars } from '../../../../shared/src/graphql-operations'
+import { asError, isErrorLike } from '../../../../shared/src/util/errors'
+import { useObservable } from '../../../../shared/src/util/useObservable'
+import { AuthenticatedUser } from '../../auth'
+import { BreadcrumbsProps, BreadcrumbSetters } from '../../components/Breadcrumbs'
+import { PageHeader } from '../../components/PageHeader'
+import { PageTitle } from '../../components/PageTitle'
+import { CodeMonitorFields, MonitorEmailPriority } from '../../graphql-operations'
+import { fetchCodeMonitor, updateCodeMonitor } from './backend'
```
Comment (limitedmage) [inline client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:16]: These should be injected as props instead, so we can unit test this component.

File: client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx [hunk 3]
```diff
@@ -1,3 +1,115 @@
...
+
+    props.useBreadcrumb(
+        React.useMemo(
+            () => ({
+                key: 'Manage Code Monitor',
+                element: <>Manage code monitor</>,
+            }),
+            []
+        )
+    )
```
Comment (limitedmage) [inline client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:64]: Should the breadcrumb be instead the code monitor's name? Or an extra level of breadcrumb with the name? (What is our usual pattern for this?)

File: client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx (hunk 1 above)
Comment (limitedmage) [inline client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:105]: When clicking "Save", the message says `Successfully created monitor`, we probably want to update this in the edit scenario

**Review by limitedmage — APPROVED** [review]
Looks good! Found some issues but they are fairly minor/can be fixed in a follow up PR. 
What are the plans for testing this component?

**Comment by attfarhan** [comment]
Testing: I'll add unit tests in a separate PR!

File: client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx (hunk 1 above)
Comment (attfarhan) [reply client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:105]: The notice will be removed actually and we'll just redirect to the listing page.

Comment (attfarhan) [reply client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:105]: Yeah I have this down as a bug, will fix separately.

File: client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx (hunk 2 above)
Comment (attfarhan) [reply client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx:16]: Will do in my next PR, which will add unit tests.

**Pull request merged**